chainconfig/chainconfig_routine.go

```diff
@@ -84,9 +84,22 @@ func (cc *ChainConfigRoutine) run() {
 				cc.logger.Error("Failed to create ChainConfigClient", "err", err)
 			} else {
 				// NOTE: errors are logged by the client, no need to log again
-				_ = chainConfigClient.VoteToEnablePendingFeatures(cc.buildNumber)
+				chainConfigClient.VoteToEnablePendingFeatures(cc.buildNumber)
 			}
 		}
 		time.Sleep(time.Duration(cc.cfg.EnableFeatureInterval) * time.Second)
 	}
 }
+
+func (cc *ChainConfigRoutine) init() {
+	if cc.node.IsValidator() {
+		dappClient := client.NewDAppChainRPCClient(cc.chainID, cc.cfg.DAppChainWriteURI, cc.cfg.DAppChainReadURI)
+		chainConfigClient, err := NewChainConfigClient(dappClient, cc.address, cc.signer, cc.logger)
+		if err != nil {
+			cc.logger.Error("Failed to create ChainConfigClient", "err", err)
+		} else {
+			// NOTE: errors are logged by the client, no need to log again
+			chainConfigClient.SetBuildNumber(cc.buildNumber)
```

Comment:
This should be done once in `run()`, I don't think we need this separate `init()` function.